pkg/controller.v1/common/job.go

```diff
@@ -455,3 +455,10 @@ func (jc *JobController) CleanupJob(runPolicy *apiv1.RunPolicy, jobStatus apiv1.
 func (jc *JobController) calcPGMinResources(minMember int32, replicas map[apiv1.ReplicaType]*apiv1.ReplicaSpec) *corev1.ResourceList {
 	return CalcPGMinResources(minMember, replicas, jc.PriorityClassLister.Get)
 }
+
+func (jc *JobController) ManagedByExternalController(rp apiv1.RunPolicy) *string {
```

Comment:
It is probably better to use `rp *apiv1.RunPolicy` to avoid copying the replica, or `managedBy *string` as suggested in https://github.com/kubeflow/training-operator/pull/2203/files#r1756807676